modules/administration-guide/pages/configuring-che-storage-size.adoc

```diff
@@ -0,0 +1,38 @@
+:_content-type: CONCEPT
+:description: Configuring the storage size in {prod-short}
+:keywords: administration guide, configuring, {prod-id-short}, storage size, PVC size, pvc
+:navtitle: Configuring the storage size in {prod-short}
+:page-aliases: installation-guide:configuring-che-storage-size.adoc
+
+[id="configuring-{prod-id-short}-workspace-pvc-size"]
+= Configuring the storage size in {prod-short}
+
+The persistent volume claim (PVC) size can be configured when using the `per-user` or `per-workspace` storage strategies. PVC sizes in the Che Cluster Custom Resource must be specified in the format of a https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/quantity/[{kubernetes} resource quantity]. For more details on the available storage strategies, see xref:configuring-che-storage-strategy.adoc[]
+
+.Default persistent volume claim sizes:
+
+* `per-user`: `10Gi`
+* `per-workspace`: `5Gi`
+
+.Procedure
+
+Use CheCluster Custom Resource definition to define the persistent volume claim size for the desired storage strategy:
+
+Set the appropriate `claimSize` field for the desired storage strategy in the Che Cluster Custom Resource. To set this field prior to installing {prod-short}, see xref:using-chectl-to-configure-the-checluster-custom-resource-during-installation.adoc[], otherwise consult xref:using-the-cli-to-configure-the-checluster-custom-resource.adoc[].
+
+[source,yaml,subs="+quotes,+attributes"]
+----
+spec:
+  devEnvironments:
+    storage:
+      pvc:
+        # the claimSize field that will be used depends on the selected pvcStrategy
+        # in this example, a persistent volume claim size of 15Gi will be used as the selected pvcStrategy is 'per-user'
+        pvcStrategy: 'per-user'
+        perUserStrategyPvcConfig:
+          # keep blank unless you need to use a non default persistent volume claim size
+          claimSize: 15Gi
+        perWorkspaceStrategyPvcConfig:
+          # keep blank unless you need to use a non default persistent volume claim size
+          claimSize: 2Gi      
+----
```

Comment:
Rather than use in-block comments, AsciiDoc allows us to mark lines with numbered callouts and then add legends for the callout numbers under the block:
```suggestion
[source,yaml,subs="+quotes,+attributes"]
----
spec:
  devEnvironments:
    storage:
      pvc:
        pvcStrategy: '__<strategy_name>__'  <1>
        perUserStrategyPvcConfig: <2>
        perWorkspaceStrategyPvcConfig:  <3>
          claimSize: 15Gi
----
<1> Enter the storage strategy: `per-user` or * `per-workspace` or `ephemeral`.
<2> Specify a claim size or leave blank for the default value. 
<3> Specify a claim size or leave blank for the default value. 
```
Please double-check whether the callout numbers in my suggestion above are on the correct lines. If not, then please edit this suggestion and move the callout numbers to the relevant lines.

Reply:
> Please double-check whether the callout numbers in my suggestion above are on the correct lines.

They are correct :) Though, since callout `<2>` and `<3>` are the same, maybe we can get rid of callout `<3>` and reuse callout `<2>` twice? 

> Is there a `ephemeralStrategyPvcConfig` line? Or does `ephemeral` strategy not need such a line? Either way, please add this info explicitly.

There is no `ephemeralStrategyPvcConfig` (as ephemeral implies non-persistent storage, i.e. all stored files will be deleted when the workspace is stopped). I added a note regarding this in the first callout:

```asciidoc
<1> Select the storage strategy: `per-user` or * `per-workspace` or `ephemeral`. Note: the `ephemeral` storage strategy does not use persistent storage, therefore you cannot configure its storage size or other PVC-related attributes.
```

Not sure if this info should be a part of the callout, or somewhere else. I think having it as part of the callout makes sense, however

> 
> Do these fields support only `Gi`? Do they support other units? If so, please let me know in the comments below.

Other units are supported as they are [Kubernetes resource quantities](https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/quantity/). I find the documentation Kubernetes provides to be a little confusing, however, so I consulted some other explanations of the units available for storage sizes. 

[This page](https://medium.com/swlh/understanding-kubernetes-resource-cpu-and-memory-units-30284b3cc866) gives a nice breakdown, see the "Memory Units" section.

## To summarize the storage size units:
Storage sizes are defined in bytes. There are 4 different ways of specifying the number of bytes for the storage size.

### Explicit Integer
You can define the size as an explicit integer to specify the exact number of bytes, e.g. `128974848`

### Exponent
You can use exponent notation using `e`, for example: `129e6`, this results in some rounding and is an approximation of the value you enter. 

### Fixed-point suffix

The following fixed-point suffixes are available, and are similar to storage units we typically use on consumer hardware (e.g. hardrives, flash drives, etc): `E, P, T, G, M, K`

- E = Exabyte. 1E = 1,000,000,000,000,000,000 bytes
- P = Petabyte. 1P = 1,000,000,000,000,000 bytes
- T = Terabyte. 1T = 1,000,000,000,000 bytes
- G = Gigabyte. 1G = 1,000,000,000 bytes
- M = Megabyte. 1M = 1,000,000 bytes
- K = Kilobyte. 1K = 1,000 bytes

### Binary, Power-of-Two Suffix

The following power-of-two suffixes are available:  `Ei, Pi, Ti, Gi, Mi, Ki` (it seems you can also add a trailing B, e.g. `EiB, PiB, TiB, GiB, MiB, KiB`, though this detail can probably be ignored for simplicity)

- Ei = EiB = Exbibyte. 1Ei = 2⁶⁰ = 1,152,921,504,606,846,976 bytes
- Pi = PiB = Pebibyte. 1Pi = 2⁵⁰ = 1,125,899,906,842,624 bytes
- Ti = TiB = Tebibyte. 1Ti = 2⁴⁰ = 1,099,511,627,776 bytes
- Gi = GiB = Gibibyte. 1Gi = 2³⁰ = 1,073,741,824 bytes
- Mi = MiB = Mebibyte. 1Mi = 2²⁰ = 1,048,576 bytes
- Ki = KiB = Kibibyte. 1Ki = 2¹⁰ = 1,024 bytes


### Thoughts on storage size units:
- In practice, I don't think users will use explicit integers or exponent notation for specifying sizes. The most commonly used size unit that I've seen is binary power-of-two suffix notation, though fixed-point suffix notation seems reasonable to use as it is more commonly seen on consumer storage hardware. 
- Since the [official Kubernetes documentation](https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/quantity/) on resource quantity is a bit confusing IMO, it may be worth having the above information explicitly stated in the Che docs? Either on the configuring storage size page, or as a separate docs page.